source/linux/Foundational_Components/U-Boot/UG-Network-K3.rst

```diff
@@ -50,48 +50,52 @@ binaries files to be sent over TFTP are listed in the table below.
 
-.. ifconfig:: CONFIG_part_variant in ('AM64X')
+.. ifconfig:: CONFIG_part_variant in ('J722S')
 
-  If using ISC dhcpd an example host entry would look like this:
+   Build tiboot3.bin using ``j722s_evm_r5_ethboot_defconfig``. Build tispl.bin and u-boot.img using ``j722s_evm_a53_ethboot_defconfig``. For instructions to build the bootloader images please refer to :ref:`Build-U-Boot-label`.
 
-  .. code-block:: text
+   .. important::
 
-      subnet 10.0.0.0 netmask 255.0.0.0
-      {
-        range dynamic-bootp 10.0.0.2 10.0.0.16;
-        if substring (option vendor-class-identifier, 0, 16) = "TI K3 Bootp Boot"
-        {
-          filename "tiboot3.bin";
-        } elsif substring (option vendor-class-identifier, 0, 20) = "AM64X U-Boot R5 SPL"
-        {
-          filename "tispl.bin";
-        } elsif substring (option vendor-class-identifier, 0, 21) = "AM64X U-Boot A53 SPL"
-        {
-          filename "u-boot.img";
-        }
+      - Ethernet RGMII boot is supported over RGMII on J722s SoC.
+      - CPSW PHYs should be strapped as per ROM's expectation described in part's TRM.
+      - When the link info Bootmode pin is enabled, this means no auto-negotiation or reading of the Ethernet PHY is needed since the ROM will assume the link is up at 1Gbps, full duplex mode.
+      - Disable link info Bootmode pin so that ROM can identify the PHY and establishes link with the supported speed and duplex mode.
 
-        range 10.0.0.17 10.0.0.25;
-        default-lease-time 60000;
-        max-lease-time 720000;
-        next-server 10.0.0.1;
-      }
+.. ifconfig:: CONFIG_part_variant in ('AM62PX')
 
-.. ifconfig:: CONFIG_part_variant in ('AM62X')
+   Build tiboot3.bin using ``am62px_evm_r5_ethboot_defconfig``. Build tispl.bin and u-boot.img using ``am62px_evm_a53_ethboot_defconfig``. For instructions to build the bootloader images please refer to :ref:`Build-U-Boot-label`.
+
+   .. important::
+
+      - Ethernet RGMII boot is supported over RGMII on AM62px SoC.
+      - CPSW PHYs should be strapped as per ROM's expectation described in part's TRM.
+      - When the link info Bootmode pin is enabled, this means no auto-negotiation or reading of the Ethernet PHY is needed since the ROM will assume the link is up at 1Gbps, full duplex mode.
+      - Disable link info Bootmode pin so that ROM can identify the PHY and establishes link with the supported speed and duplex mode.
+
+.. ifconfig:: CONFIG_part_variant in ('J721S2')
+
+   Build tiboot3.bin using ``am68_evm_r5_ethboot_defconfig``. Build tispl.bin and u-boot.img using ``am68_evm_a72_ethboot_defconfig``. For instructions to build the bootloader images please refer to :ref:`Build-U-Boot-label`.
+
+   .. important::
+
+      - Ethernet RGMII boot is supported over RGMII on J721S2 SoC.
+
+.. ifconfig:: CONFIG_part_variant in ('AM64X','AM62X','J722S','AM62PX','J721S2')
 
   If using ISC dhcpd an example host entry would look like this:
 
-  .. code-block:: text
+  .. parsed-literal::
```

Comment:
Alright, this can kind of work, but there are issues.

1. Ifconfig is essentially a mask. This section will still be parsed on other platforms and will result in a replacement error if they do not have the replacement definition. Right now you've defined the variable for all machines that include this file, so it's fine. You can remove that ifconfig mask as it's not doing anything.

2. Parsed literals will not be escaped. Any valid RST will be parsed. This example is fine right now. Any additions will have to take this into account.

This is why I was pushing more towards defining a table and user template here. I'll let others weigh in.